common/getmmdb.sh

```diff
@@ -1,13 +1,61 @@
 #!/bin/sh -e 
 echo "2018-09-21"
 
-mkdir -p /usr/share/GeoIP/
-wget -N http://geolite.maxmind.com/download/geoip/database/GeoLite2-City.tar.gz
-tar xvzf GeoLite2-City.tar.gz
-rm GeoLite2-City.tar.gz
+get_geoip_db () {
+    # Fetches specified EDITION_ID db from maxmind using
+    # active LICENSE_KEY.
+    # 
+    # This gets the latest version of the db. To get db corresponding to a
+    # particular date, add the following URL paramater:
+    #
+    #   date=YYYYMMDD (e.g. 20200107)
+    #
+    # As described in [2] below, this requires an active user account
+    # and an associated LICENSE_KEY (free).
+    #
+    # References:
+    #
+    # [1] https://dev.maxmind.com/geoip/geoipupdate/#Direct_Downloads
+    # [2] https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-geolite2-databases/
 
-wget -N http://geolite.maxmind.com/download/geoip/database/GeoLite2-ASN.tar.gz
-tar xvzf GeoLite2-ASN.tar.gz
-rm GeoLite2-ASN.tar.gz
+    EDITION_ID=$1
+    SUFFIX=$2
+    LICENSE_KEY=$3
 
-mv */*.mmdb /usr/share/GeoIP
+    BASE_URL="https://download.maxmind.com/app/geoip_download"
+    wget "$BASE_URL?edition_id=$EDITION_ID&license_key=$LICENSE_KEY&suffix=$SUFFIX" \
+        -O $EDITION_ID.$SUFFIX
+}
+
+
+main () {
```

Comment:
this should check if the license key is set before trying to download files.

Reply:
👍  I've added the check in the next commit.